modules/data-streams/src/test/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleErrorStoreTests.java

```diff
@@ -68,4 +69,11 @@ public void testGetAllIndicesIsASnapshotViewOfTheStore() {
             containsInAnyOrder(Stream.iterate(2, i -> i + 1).limit(8).map(i -> "test" + i).toArray(String[]::new))
         );
     }
+
+    public void testRecordedErrorIsMaxOneThousandChars() {
+        NullPointerException exceptionWithLongMessage = new NullPointerException(randomAlphaOfLength(2000));
+        errorStore.recordError("test", exceptionWithLongMessage);
+        assertThat(errorStore.getError("test"), is(notNullValue()));
```

Comment:
Nit: I would prefer to have this check if the `exceptionWithLongMessage` starts with `errorStore.getError("test")`. However, if I am not mistaken the available matchers do not support something like a substring, so we cannot nicely do this.

We could alternatively, create an error message of 1000 characters and append some extra characters and do an equality comparison on the first part. 

Again this is a nit. I am just thinking out loud.

Reply:
Thanks for the review Mary. IMO what we actually store in the error store would be an implementation detail. This test is solely interested in the size of the stored message.

Reply:
I think the actual stored error would be in an exception stack trace format (starting with the FQ class name) so starts with wouldn't match the actual message.